framework/db/transaction_manager.py

```diff
@@ -438,6 +441,43 @@ def GetByIDAsDict(self, trans_id, target_id=None):
         return self.DeriveTransactionDict(transaction_obj, include_raw_data=True)
 
     @target_required
+    def GetHrtResponse(self, filter_data, trans_id, target_id=None):
+        transaction_obj = self.db.session.query(models.Transaction).filter_by(target_id=target_id, id=trans_id).first()
+
+        # Data validation
+        languages = ['bash']  # Default script language is set to bash.
+        if filter_data.get('language'):
+            languages = map(lambda x: x.strip(), filter_data['language'][0].split(','))
+
+        if filter_data.get('proxy'):
+            proxy = None if filter_data['proxy'][0] == '' else filter_data['proxy'][0]
+
+        if filter_data.get('search_string'):
+            search_string = None if filter_data['search_string'][0] == '' else filter_data['search_string'][0]
+
+        if filter_data.get('data'):
+            data = None if filter_data['data'][0] == '' else filter_data['data'][0]
+
+        # If target not found. Raise error.
+        if not transaction_obj:
+            raise InvalidTransactionReference("No transaction with %s exists" % str(trans_id))
+
+        raw_request = transaction_obj.raw_request
+
+        try:
+            hrt_obj = HttpRequestTranslator(
+                request=raw_request,
+                languages=languages,
+                proxy=proxy,
```

Comment:
`proxy` might be undefined at that point if `filter_data` does not contain `'proxy'`. Same for the other parameters.